### DISCUSSION TITLE
Is my approach to force my customized function receive all rows as parameter 100% always available?

### DISCUSSION THREAD

**Discussion by Alex-Cheng** [ClickHouse/ClickHouse#34080] [Q&A] [answered]
I have a customized function which needs to get all rows as input because it needs to look around all rows when processing a row.  I find a way to make it, that is adding the following SETTINGS clause at the end of query. 

`SETTINGS max_block_size = 100000000, max_threads=1, preferred_block_size_bytes=0`

Although in practice the approach works, but I am wondering if the approach will 100% always work? Can you give me an answer? Thank you.

**Comment by alexey-milovidov** [comment, marked as answer]
It is not guaranteed to work - the setting `max_block_size` recommends the maximum block size for storages, but some storages will still return smaller blocks. For example, StripeLog and Memory engines return blocks precisely as they were inserted. MergeTree engine can return multiple smaller blocks if there are multiple partitions. The same applies to Merge, Buffer, and Distributed tables.
